cmd/go-vcs/go-vcs.go

```diff
@@ -309,23 +310,38 @@ func main() {
 		}
 
 	case "branches":
-		if len(args) != 0 {
-			log.Fatal("branches takes no arguments.")
+		if len(args) > 1 {
+			log.Fatal("branches takes 0 or 1 arguments.")
+		}
+
+		// Open using go/vcs to figure out VCS type (git, hg).
+		r := vcs2.New(".")
+		if r == nil {
+			log.Fatalln("no supported vcs found in cwd")
 		}
 
-		repo, err := vcs.Open("hg", ".")
+		repo, err := vcs.Open(r.Type().VcsType(), ".")
```

Comment:
You don't need to make this change in the upstream lib, but I thought I'd mention it anyway. VcsType should be VCSType, according to [CodeReviewComments#Initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms).